TLM/TLM/UI/SubTools/SpeedLimits/SpeedLimitsTool.cs

```diff
@@ -919,19 +922,19 @@ public SpeedLimitsTool(TrafficManagerTool mainTool)
 
             if (GlobalConfig.Instance.Main.DisplaySpeedLimitsMph) {
                 MphValue rounded = speed.ToMphRounded(MPH_STEP);
-                rounded.Mph += MPH_STEP;
+                rounded = new MphValue((ushort)(rounded.Mph + MPH_STEP));
 
                 if (rounded.Mph > UPPER_MPH) {
-                    rounded.Mph = 0;
+                    rounded = new MphValue(0);
                 }
 
                 return SpeedValue.FromMph(rounded);
             } else {
                 KmphValue rounded = speed.ToKmphRounded(KMPH_STEP);
-                rounded.Kmph += KMPH_STEP;
+                rounded = new KmphValue((ushort)(rounded.Kmph + KMPH_STEP));
```

Comment:
You can simplify the code by introducing an overloaded operator.
Add an operator to your struct:
```
readonly struct KmphValue
{
    public static KmphValue operator+(KmphValue value, ushort delta)
        => new KmphValue((ushort)(value.Kmph + delta));
}
```

Now you can do:
```
var value = new KmphValue(10);
value += 5;
```

Reply:
Added that.
For now we don't use these speed values in the game AI, only in speed limits UI. But in fact the code is ready to be used in the AI code just as well.